artifactory/commands/generic/delete.go

```diff
@@ -65,31 +63,45 @@ func (dc *DeleteCommand) Run() error {
 	return nil
 }
 
-func (dc *DeleteCommand) GetPathsToDelete() error {
+func (dc *DeleteCommand) GetPathsToDelete() (*content.ContentReader, error) {
 	rtDetails, err := dc.RtDetails()
 	if errorutils.CheckError(err) != nil {
-		return err
+		return nil, err
 	}
 	servicesManager, err := utils.CreateServiceManager(rtDetails, dc.DryRun())
 	if err != nil {
-		return err
+		return nil, err
 	}
+	temp := []*content.ContentReader{}
 	for i := 0; i < len(dc.Spec().Files); i++ {
 		deleteParams, err := getDeleteParams(dc.Spec().Get(i))
 		if err != nil {
-			return err
+			return nil, err
 		}
 
-		currentResultItems, err := servicesManager.GetPathsToDelete(deleteParams)
+		currentResultsReader, err := servicesManager.GetPathsToDelete(deleteParams)
 		if err != nil {
-			return err
+			return nil, err
+		}
+		length, err := currentResultsReader.Length()
+		if err != nil {
+			return nil, err
+		}
+		if length > 0 {
+			temp = append(temp, currentResultsReader)
+		} else {
+			currentResultsReader.Close()
 		}
-		dc.deleteItems = append(dc.deleteItems, currentResultItems...)
 	}
-	return nil
+	defer func() {
```

Comment:
It is better to add the close of the readers with "ddefer" right after each reader is returned. There's probably no need for this loop. The way it is done now, if an error is returned before the function ends, these readers can be left open.
Also, there's another currentResultsReader.Close() above, in case the length is 0. I'm not sure we need to handle this case seperately (and without a defer).